test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java

```diff
@@ -279,55 +281,118 @@ public void testGenome() throws Exception {
         Assert.assertEquals("1", "" + assemblyIndex.keyProps.get("contigs"));
         System.out.println("*** end testGenome***");
     }
+
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testMultiTypeSearchValidation1() throws Exception {
+
+        // null list of types
+        Set<GUID> guids = indexStorage.searchIds(null,
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+    }
+
+    @Test
+    public void testMultiTypeSearchValidation2() throws Exception {
+
+        // empty list
```

Comment:
Will this test pass if it runs first? It looks like it expects that at least one object is indexed, but I think all the tests index their own objects.

Really we should clear the database between every test so that tests are ensured to be independent.

Reply:
If I correctly understood how prepare() is implemented, we have at least 2 objects being indexed before the tests are executed.

Reply:
I just reread prepare() and I'm not seeing any indexing that happens as part of that method. There's indexObject calls embedded in an anonymous subclass of ObjectLookupProvider that's declared in prepare() but that won't be executed as part of the prepare method(). Am I missing something?

Reply:
You were right and I was being lazy. I fixed the side effects that were affecting the tests. The database is now being cleared.

Reply:
Actually, I was right. When a genome01 object is created in a test, the lookup provider object's resolveRefs method gets called in prepare() and this populates the index with two more referenced objects

Reply:
But that doesn't help if a test that doesn't index any objects is run first, right? It'll still fail because no objects are indexed. The lookup provider just adds more objects the first time a test indexes a genome, so it doesn't affect tests that don't index anything but expect things to be indexed.

Reply:
testMultiTypeSearchValidation2() indexes a genome object which causes that awful implementation in prepare to index two more objects.

Reply:
Well it does now. When this comment chain started it didn't.